R/recipe.R

```diff
@@ -188,7 +196,10 @@ recipe.formula <- function(formula, data, ...) {
   # check for minus:
   f_funcs <- fun_calls(formula)
   if (any(f_funcs == "-")) {
-    rlang::abort("`-` is not allowed in a recipe formula. Use `step_rm()` instead.")
+    cli::cli_abort(c(
+      "x" = "{.code -} is not allowed in a recipe formula.",
+      "i" = "Use {.fn recipes::step_rm} instead."
```

Comment:
```suggestion
      "i" = "Use {.fn step_rm} instead."
```

I think, since this is in the package where the error is thrown from, we're fine to not include namespace. We refer to `bake()`, `prep()`, etc elsewhere throughout this PR without namespacing.

Reply:
I would like to keep this one, as it provides a clickable link, which I think is a nice touch since we are asking them to use a function they are not currently using

<img width="801" alt="Screenshot 2023-11-01 at 8 59 24 AM" src="https://github.com/tidymodels/recipes/assets/14034784/ee03c11d-40b6-44ad-a128-557d11c3057b">

But I'm gonna modify it by using `{.help [{.fun step_rm}](recipes::step_rm)}` such that `recipes::` doesn't show in the error message

![Uploading Screenshot 2023-11-01 at 9.14.58 AM.png…]()